creator-node/test/snapbackSM.test.js

```diff
@@ -55,53 +58,75 @@ describe('test sync queue', function () {
+      recurringSyncIds.push(id)
     }
 
     // setup manual syncs
-    const manualSyncIds = new Set()
+    const manualSyncIds = []
     for (let i = 0; i < 3; i++) {
       const { id } = await snapback.enqueueManualSync({
         userWallet: constants.userWallet,
         secondaryEndpoint: constants.secondaryEndpoint,
         primaryEndpoint: constants.primaryEndpoint,
-        priority: SyncPriority.High,
         syncType: SyncType.Manual
       })
-      manualSyncIds.add(id)
+      manualSyncIds.push(id)
     }
 
-    // Verify we complete manual jobs first
-    let jobIds = (await snapback.getSyncQueueJobs()).pending.map(job => job.id)
-    let lastRemainingRecurringCount = 0
-
-    while (jobIds.length) {
-      const remainingManualCount = jobIds.filter(id => manualSyncIds.has(id)).length
-      const remainingRecurringCount = jobIds.filter(id => recurringSyncIds.has(id)).length
-
-      // We know we processed a recurring job before a manual job
-      // if there are still manual jobs left to process but we have fewer
-      // unprocessed recurring jobs remaining than we did last iteration through the loop
-      const didProcessRecurringBeforeManual = remainingManualCount > 0 && remainingRecurringCount < lastRemainingRecurringCount
-
-      if (didProcessRecurringBeforeManual) {
-        assert.fail('Should have done all manual high priority syncs first')
+    const totalJobsAddedCount = recurringSyncIds.length + manualSyncIds.length
+
+    let syncQueueJobs = await snapback.getSyncQueueJobs(true)
+    let [
+      manualWaitingJobIDs,
+      recurringWaitingJobIDs
+    ] = [
+      syncQueueJobs.manualWaiting.map(job => job.id),
+      syncQueueJobs.recurringWaiting.map(job => job.id)
+    ]
+
+    // Keep polling until no waiting jobs remaining
+    // Set polling timeout to (2 * totalJobsAddedCount * NodeResponseDelayMs) to ensure jobs are being processed in a timely manner
+    let totalPollingTime = 0
+    while (manualWaitingJobIDs.length || recurringWaitingJobIDs.length) {
+      await utils.timeout(NodeResponseDelayMs)
+
+      totalPollingTime += NodeResponseDelayMs
+      if (totalPollingTime > (2 * totalJobsAddedCount * NodeResponseDelayMs)) {
+        throw new Error('Snapback failed to process sync queue jobs in a timely manner')
       }
 
-      lastRemainingRecurringCount = remainingRecurringCount
-
-      await utils.timeout(500)
-      jobIds = (await snapback.getSyncQueueJobs()).pending.map(job => job.id)
+      syncQueueJobs = await snapback.getSyncQueueJobs(true);
+      [
+        manualWaitingJobIDs,
+        recurringWaitingJobIDs
+      ] = [
+        syncQueueJobs.manualWaiting.map(job => job.id),
+        syncQueueJobs.recurringWaiting.map(job => job.id)
+      ]
     }
+
+    assert.strictEqual(manualWaitingJobIDs.length, 0)
+    assert.strictEqual(recurringWaitingJobIDs.length, 0)
+
+    // TODO - not sure why this isn't working, bull is not returning any jobs as completed 🤷‍♂️
```

Comment:
what's the deal here? so bull doesn't store this state?

Reply:
@dmanjunath resolving for now